### PR TITLE
fix crossing

### LINKED ISSUE #3107: Should not allow user to cross order book
1. Need to calculate spread value,

image

2.
don't allow user to add a ask price under best bid.
don't allow user to add a bid over best ask.

notice below, the issue the best ask is lower than best bid.

image

image

### PR DESCRIPTION
**Pull request by phoebemirman**
closes https://github.com/AugurProject/augur/issues/3107

### CODE REVIEW

File: packages/augur-ui/src/modules/market/containers/trading-form.ts [hunk 1]
```diff
@@ -20,17 +20,30 @@ import {
...
-  const hasFunds = !!state.loginAccount.balances.eth && !!state.loginAccount.balances.dai
+  const hasFunds = !!state.loginAccount.balances.eth && !!state.loginAccount.balances.dai;
+
+  const selectedOutcomeId = (ownProps.selectedOutcomeId !== undefined && ownProps.selectedOutcomeId !== null) ? ownProps.selectedOutcomeId : market.defaultSelectedOutcomeId;
+  let outcomeOrderBook = ownProps.market.orderBook[selectedOutcomeId];
+  if (ownProps.initialLiquidity) {
+    outcomeOrderBook = formatOrderBook(outcomeOrderBook);
+  }
+
+  const cumulativeOrderBook = orderAndAssignCumulativeShares(
```
Comment (bthaile) [inline packages/augur-ui/src/modules/market/containers/trading-form.ts:37]: cumulativeOrderBook isn't referenced in this file, can be removed?

File: packages/augur-ui/src/modules/market/containers/trading-form.ts [hunk 2]
```diff
@@ -20,17 +20,30 @@ import {
...
+import { formatOrderBook } from 'modules/create-market/helpers/format-order-book';
 
 const mapStateToProps = (state, ownProps) => {
   const { authStatus, loginAccount } = state;
 
-  const hasFunds = !!state.loginAccount.balances.eth && !!state.loginAccount.balances.dai
+  const hasFunds = !!state.loginAccount.balances.eth && !!state.loginAccount.balances.dai;
+
+  const selectedOutcomeId = (ownProps.selectedOutcomeId !== undefined && ownProps.selectedOutcomeId !== null) ? ownProps.selectedOutcomeId : market.defaultSelectedOutcomeId;
+  let outcomeOrderBook = ownProps.market.orderBook[selectedOutcomeId];
```
Comment (bthaile) [inline packages/augur-ui/src/modules/market/containers/trading-form.ts]: orderbook isn't on the market, use the order book selector, select-order-book.ts

**Review by bthaile — APPROVED** [review]
lgtm

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🍭
 To view it, be sure to [join Augur Project](https://dashboard.featurepeek.com/login) on FeaturePeek.

 This link will always point to this PR's latest changes: https://peek.run/654h1r9

**Pull request merged**